mcp/src/minecraft/com/elementfx/tvp/ad/item/crafting/GoblinCraftingManager.java

```diff
@@ -0,0 +1,198 @@
+package com.elementfx.tvp.ad.item.crafting;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import net.minecraft.block.Block;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.InventoryCrafting;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.CraftingManager;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.ShapedRecipes;
+import net.minecraft.item.crafting.ShapelessRecipes;
+import net.minecraft.world.World;
+
+public class GoblinCraftingManager
+{
+	 /** The static instance of this class */
+    private static final GoblinCraftingManager INSTANCE = new GoblinCraftingManager();
+    private final List<IRecipe> recipes = Lists.<IRecipe>newArrayList();
+
+    /**
+     * Returns the static instance of this class
+     */
+    public static GoblinCraftingManager getInstance()
+    {
+        return INSTANCE;
+    }
+
+    protected GoblinCraftingManager()
+    {
+    	Collections.sort(this.getRecipeList(), new Comparator<IRecipe>()
+        {
+            public int compare(IRecipe p_compare_1_, IRecipe p_compare_2_)
+            {
+                return p_compare_1_ instanceof ShapelessRecipes && p_compare_2_ instanceof ShapedRecipes ? 1 : (p_compare_2_ instanceof ShapelessRecipes && p_compare_1_ instanceof ShapedRecipes ? -1 : (p_compare_2_.getRecipeSize() < p_compare_1_.getRecipeSize() ? -1 : (p_compare_2_.getRecipeSize() > p_compare_1_.getRecipeSize() ? 1 : 0)));
+            }
+        });
+    }
+    /**
+     * Adds a shaped recipe to the games recipe list.
+     */
+    public ShapedRecipes addRecipe(ItemStack stack, Object... recipeComponents)
```

Comment:
Can we have a parent `CustomCraftingManager` class which contains all of these reused functions in XCraftingManager?